cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -396,67 +448,75 @@ def create_rna_dna_sample_case_map(self, rna_case: models.Family) -> Dict[str, D
+        rna_dna_sample_case_map: Dict[str, Dict[str, List[str]]] = {}
+        for link in rna_case.links:
+            self._map_rna_sample(
+                rna_sample=link.sample, rna_dna_sample_case_map=rna_dna_sample_case_map
             )
         return rna_dna_sample_case_map
 
-    def _add_rna_sample(
-        self, rna_sample: models.Sample, rna_dna_sample_case_map: Dict[str, Dict[str, list]]
-    ) -> Dict[str, Dict[str, list]]:
-        """Adds an RNA sample and its matching DNA sample, and cases."""
-        dna_sample: models.Sample = self._link_rna_sample_to_dna_sample(
+    def _map_rna_sample(
+        self, rna_sample: models.Sample, rna_dna_sample_case_map: Dict[str, Dict[str, List[str]]]
+    ) -> Dict[str, Dict[str, List[str]]]:
+        """Create a dictionary of all DNA samples, and their related cases, related to a RNA sample."""
+        dna_sample: models.Sample = self._map_dna_samples_related_to_rna_sample(
             rna_sample=rna_sample, rna_dna_sample_case_map=rna_dna_sample_case_map
         )
-        self._add_dna_cases_to_dna_sample(
+        self._map_dna_cases_to_dna_sample(
             dna_sample=dna_sample,
             rna_dna_sample_case_map=rna_dna_sample_case_map,
             rna_sample=rna_sample,
         )
         return rna_dna_sample_case_map
 
-    def _link_rna_sample_to_dna_sample(
-        self, rna_sample: models.Sample, rna_dna_sample_case_map: Dict[str, Dict[str, list]]
+    def _map_dna_samples_related_to_rna_sample(
+        self, rna_sample: models.Sample, rna_dna_sample_case_map: Dict[str, Dict[str, List[str]]]
     ) -> models.Sample:
-
+        """Maps an RNA sample to a DNA sample based on subject id."""
         if not rna_sample.subject_id:
             raise CgDataError(
-                f"Failed on RNA sample {rna_sample.internal_id} as subject_id field is empty"
+                f"Failed to link RNA samepl {rna_sample.internal_id} to dna samples - subject_id field is empty"
             )
 
-        subject_id_samples: Query = self.status_db.samples_by_subject_id(
-            customer_id=rna_sample.customer.internal_id,
-            subject_id=rna_sample.subject_id,
-            is_tumour=rna_sample.is_tumour,
-        )
-        nr_of_subject_id_dna_samples: int = len(subject_id_samples.all())
-        if nr_of_subject_id_dna_samples != 2:
-            raise CgDataError(
-                f"Failed to upload files for RNA case: unexpected number of DNA sample matches for subject_id: {rna_sample.subject_id}. Number of matches: {nr_of_subject_id_dna_samples} "
+        subject_id_samples: List[models.Sample] = self.validate_number_of_dna_samples_by_subject_id(
+            self.status_db.samples_by_subject_id(
+                customer_id=rna_sample.customer.internal_id,
+                subject_id=rna_sample.subject_id,
+                is_tumour=rna_sample.is_tumour,
             )
```

Comment:
Perhaps this should be split into 2 lines. One for getting the subject_id_samples and then one for calling the validate function. Then validate does not have to return anything and can be a pure function (i.e. just validating and not returning anything)